lib/puppet/provider/user/useradd.rb

```diff
@@ -186,18 +186,43 @@ def addcmd
     cmd << @resource[:name]
   end
 
+  def modifycmd(param, value)
+    if @resource.forcelocal?
+      cmd = [command(param.to_s =~ /password_.+_age/ ? :localpassword : :localmodify)]
```

Comment:
This regex won't match ':password_warn_days'. Do we not call modify cmd on those?

Reply:
Thank you for catching this bug @ericldelaney!

It looks like this has been broken for at least  a year so far https://github.com/puppetlabs/puppet/blob/4d96086ac88b34f3ed78d61a14e246016db22b17/lib/puppet/provider/nameservice/objectadd.rb#L16

It was introduced in March 2017 with https://github.com/puppetlabs/puppet/commit/31089218e84e898e4b2388bf1ed98e08bf892968

I've opened https://tickets.puppetlabs.com/browse/PUP-8381 to fix this one.

@treydock we decided that we shouldn't block this PR because of that bug. We can fix that separately.